examples/make_ext_smry.cpp

```diff
@@ -91,25 +91,41 @@ int main(int argc, char **argv) {
 
     auto lap0 = std::chrono::system_clock::now();
 
+    int num_esmry = argc-argOffset;
+    std::vector<bool> status(num_esmry, false);
+
     #pragma omp parallel for
-    for (int f = argOffset; f < argc; f ++){
-        std::filesystem::path inputFileName = argv[f];
+    for (int f = 0; f < num_esmry; f ++){
+        std::filesystem::path inputFileName = argv[f + argOffset];
 
         std::filesystem::path esmryFileName = inputFileName.parent_path() / inputFileName.stem();
         esmryFileName = esmryFileName += ".ESMRY";
 
         if (Opm::EclIO::fileExists(esmryFileName) && (force))
             remove (esmryFileName);
 
-        Opm::EclIO::ESmry smryFile(argv[f]);
-        if (!smryFile.make_esmry_file()){
-            std::cout << "\n! Warning, smspec already have one lod file, existing kept use option -f to replace this" << std::endl;
+        std::unique_ptr<Opm::EclIO::ESmry> smryfile;
+
+        try {
+            std::unique_ptr<Opm::EclIO::ESmry> smry_tmp = std::make_unique<Opm::EclIO::ESmry> ( argv[f + argOffset] );;
+            smryfile = std::move(smry_tmp);
+        } catch (...) {
+            std::cout << "\n! Warning, could not open summary file " + std::string(argv[f + argOffset]) + "\n";
+        }
+
+        if ( smryfile != nullptr){
+            status[f] = smryfile->make_esmry_file();
+            if (!status[f])
+                std::cout << "\n! Warning, smspec already have one esmry file, existing kept use option -f to replace this" << std::endl;
         }
```

Comment:
I think you could probably write this section as
```C++
try {
    Opm::EclIO::ESmry smry{ argv[f + argOffset] };
    status[f] = smry.make_esmry_file();
    if (! status[f]) {
        std::cerr << "\n! Warning, smspec already have one esmry file, existing kept use option -f to replace this\n";
    }
}
catch (...) {
    std::cerr << "\n! Warning, could not open summary file " << argv[f + argOffset] << '\n';
}
```
in order to avoid having to deal with pointers.

Reply:
happy to update, thanks.